File: src - c++/Esercizi/SommaLivello/SommaLivello.cpp
//Metodo privato

void AlberoBinario::_SommaLivello(const PNodo& n, unsigned int i) {
  if (n) {
    n->el += i;
    _SommaLivello(n->sin, i+1);
    _SommaLivello(n->des, i+1);
  }    
}

//Metodo pubblico

void AlberoBinario::SommaLivello() {
  _SommaLivello(root, 1);
}

File: src - java/Esercizi/SommaLivello/SommaLivello.java
	private void _sommaLivello(Nodo n, int i) {
		if (n != null) {
			n.elem += i;
			_sommaLivello(n.sin, i + 1);
			_sommaLivello(n.des, i + 1);
		}
	}

	public void sommaLivello() {
		_sommaLivello(root, 1);
	}